pkg/kademlia/dialer_test.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package kademlia_test
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap/zaptest"
+	"golang.org/x/sync/errgroup"
+
+	"storj.io/storj/internal/testcontext"
+	"storj.io/storj/internal/testplanet"
+	"storj.io/storj/pkg/kademlia"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storj"
+)
+
+func TestDialer(t *testing.T) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 4, UplinkCount: 3,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		time.Sleep(2 * time.Second)
+		expectedKademliaEntries := 1 + len(planet.Satellites) + len(planet.StorageNodes)
+
+		// TODO: also use satellites
+		peers := planet.StorageNodes
+
+		{ // Ping
```

Comment:
Maybe separate methods for those 3 groups would be cleaner? Or better comment what first and second lookup is doing.

Reply:
fixed